src/Microsoft.AspNet.Mvc.Razor/RazorView.cs

```diff
@@ -93,6 +97,7 @@ public string Path
                 throw new ArgumentNullException(nameof(context));
             }
 
+            _bufferSource = context.HttpContext.RequestServices.GetRequiredService<IRazorBufferSource>();
```

Comment:
The service is scoped to a request but view engines are singletons. So that wouldn't work.